diff -rupN linux-2.6.34-rc3-n1/arch/arm/mach-msm/acpuclock-scorpion.c linux-2.6.34-rc3-n1-ocuv/arch/arm/mach-msm/acpuclock-scorpion.c
--- linux-2.6.34-rc3-n1/arch/arm/mach-msm/acpuclock-scorpion.c	2010-04-25 00:03:39.577717770 +0930
+++ linux-2.6.34-rc3-n1-ocuv/arch/arm/mach-msm/acpuclock-scorpion.c	2010-04-25 06:04:56.000000000 +0930
@@ -66,32 +66,35 @@ struct clkctl_acpu_speed {
 
 /* core sources */
 #define SRC_RAW		0 /* clock from SPSS_CLK_CNTL */
-#define SRC_SCPLL	1 /* output of scpll 128-998 MHZ */
+#define SRC_SCPLL	1 /* output of scpll 128-1113 MHZ */
 #define SRC_AXI		2 /* 128 MHz */
 #define SRC_PLL1	3 /* 768 MHz */
 
 struct clkctl_acpu_speed acpu_freq_tbl[] = {
-	{  19200, CCTL(CLK_TCXO, 1),		SRC_RAW, 0, 0, 1000 },
-	{ 128000, CCTL(CLK_TCXO, 1),		SRC_AXI, 0, 0, 1000 },
-	{ 245000, CCTL(CLK_MODEM_PLL, 1),	SRC_RAW, 0, 0, 1000 },
-	{ 256000, CCTL(CLK_GLOBAL_PLL, 3),	SRC_RAW, 0, 0, 1000 },
-	{ 384000, CCTL(CLK_TCXO, 1),		SRC_SCPLL, 0x0A, 0, 1000 },
+	{  19200, CCTL(CLK_TCXO, 1),		SRC_RAW, 0, 0, 950 },
+	{ 128000, CCTL(CLK_TCXO, 1),		SRC_AXI, 0, 0, 950 },
+	{ 245000, CCTL(CLK_MODEM_PLL, 1),	SRC_RAW, 0, 0, 950 },
+	{ 256000, CCTL(CLK_GLOBAL_PLL, 3),	SRC_RAW, 0, 0, 950 },
+	{ 384000, CCTL(CLK_TCXO, 1),		SRC_SCPLL, 0x0A, 0, 975 },
 	{ 422400, CCTL(CLK_TCXO, 1),		SRC_SCPLL, 0x0B, 0, 1000 },
-	{ 460800, CCTL(CLK_TCXO, 1),		SRC_SCPLL, 0x0C, 0, 1000 },
-	{ 499200, CCTL(CLK_TCXO, 1),		SRC_SCPLL, 0x0D, 0, 1025 },
-	{ 537600, CCTL(CLK_TCXO, 1),		SRC_SCPLL, 0x0E, 0, 1050},
-	{ 576000, CCTL(CLK_TCXO, 1),		SRC_SCPLL, 0x0F, 0, 1050 },
-	{ 614400, CCTL(CLK_TCXO, 1),		SRC_SCPLL, 0x10, 0, 1075 },
-	{ 652800, CCTL(CLK_TCXO, 1),		SRC_SCPLL, 0x11, 0, 1100 },
-	{ 691200, CCTL(CLK_TCXO, 1),		SRC_SCPLL, 0x12, 0, 1125 },
-	{ 729600, CCTL(CLK_TCXO, 1),		SRC_SCPLL, 0x13, 0, 1150 },
-	{ 768000, CCTL(CLK_TCXO, 1),		SRC_SCPLL, 0x14, 0, 1150 },
-	{ 806400, CCTL(CLK_TCXO, 1),		SRC_SCPLL, 0x15, 0, 1175 },
-	{ 844800, CCTL(CLK_TCXO, 1),		SRC_SCPLL, 0x16, 0, 1200 },
-	{ 883200, CCTL(CLK_TCXO, 1),		SRC_SCPLL, 0x17, 0, 1225 },
-	{ 921600, CCTL(CLK_TCXO, 1),		SRC_SCPLL, 0x18, 0, 1250 },
-	{ 960000, CCTL(CLK_TCXO, 1),		SRC_SCPLL, 0x19, 0, 1250 },
-	{ 998400, CCTL(CLK_TCXO, 1),		SRC_SCPLL, 0x1A, 0, 1250 },
+	{ 460800, CCTL(CLK_TCXO, 1),		SRC_SCPLL, 0x0C, 0, 1025 },
+	{ 499200, CCTL(CLK_TCXO, 1),		SRC_SCPLL, 0x0D, 0, 1050 },
+	{ 537600, CCTL(CLK_TCXO, 1),		SRC_SCPLL, 0x0E, 0, 1050 },
+	{ 576000, CCTL(CLK_TCXO, 1),		SRC_SCPLL, 0x0F, 0, 1075 },
+	{ 614400, CCTL(CLK_TCXO, 1),		SRC_SCPLL, 0x10, 0, 1100 },
+	{ 652800, CCTL(CLK_TCXO, 1),		SRC_SCPLL, 0x11, 0, 1125 },
+	{ 691200, CCTL(CLK_TCXO, 1),		SRC_SCPLL, 0x12, 0, 1150 },
+	{ 729600, CCTL(CLK_TCXO, 1),		SRC_SCPLL, 0x13, 0, 1175 },
+	{ 768000, CCTL(CLK_TCXO, 1),		SRC_SCPLL, 0x14, 0, 1200 },
+	{ 806400, CCTL(CLK_TCXO, 1),		SRC_SCPLL, 0x15, 0, 1225 },
+	{ 844800, CCTL(CLK_TCXO, 1),		SRC_SCPLL, 0x16, 0, 1225 },
+	{ 883200, CCTL(CLK_TCXO, 1),		SRC_SCPLL, 0x17, 0, 1250 },
+	{ 921600, CCTL(CLK_TCXO, 1),		SRC_SCPLL, 0x18, 0, 1275 },
+	{ 960000, CCTL(CLK_TCXO, 1),		SRC_SCPLL, 0x19, 0, 1275 },
+	{ 998400, CCTL(CLK_TCXO, 1),		SRC_SCPLL, 0x1A, 0, 1275 },
+	{ 1036800, CCTL(CLK_TCXO, 1),		SRC_SCPLL, 0x1B, 0, 1275 },
+	{ 1075200, CCTL(CLK_TCXO, 1),		SRC_SCPLL, 0x1C, 0, 1275 },
+	{ 1113600, CCTL(CLK_TCXO, 1),		SRC_SCPLL, 0x1D, 0, 1275 },
 	{ 0 },
 };
 
@@ -117,9 +120,11 @@ static void __init acpuclk_init_cpufreq_
 		freq_table[i].index = i;
 		freq_table[i].frequency = CPUFREQ_ENTRY_INVALID;
 
-		/* Skip speeds using the global pll */
-		if (acpu_freq_tbl[i].acpu_khz == 256000 ||
-				acpu_freq_tbl[i].acpu_khz == 19200)
+		/* Define speeds that we want to skip */
+		if (/* acpu_freq_tbl[i].acpu_khz == 256000 || */
+				acpu_freq_tbl[i].acpu_khz == 19200 ||
+				acpu_freq_tbl[i].acpu_khz == 128000 || 
+				acpu_freq_tbl[i].acpu_khz == 256000)
 			continue;
 
 		vdd = acpu_freq_tbl[i].vdd;
@@ -130,8 +135,8 @@ static void __init acpuclk_init_cpufreq_
 			continue;
 		}
 
-		/* Take the fastest speed available at the specified VDD level */
-		if (vdd != acpu_freq_tbl[i + 1].vdd)
+		/* Add to the table */
+		//if (vdd != acpu_freq_tbl[i + 1].vdd)
 			freq_table[i].frequency = acpu_freq_tbl[i].acpu_khz;
 	}
 
@@ -228,7 +233,7 @@ static void scpll_set_freq(uint32_t lval
 			;
 
 		/* completion bit is not reliable for SHOT switch */
-		udelay(25);
+		udelay(15);
 	}
 
 	/* write the new L val and switch mode */
@@ -317,7 +322,8 @@ int acpuclk_set_rate(unsigned long rate,
 		/* Increase VDD if needed. */
 		if (next->vdd > cur->vdd) {
 			if (acpuclk_set_vdd_level(next->vdd)) {
-				pr_err("acpuclock: Unable to increase ACPU VDD.\n");
+/*				pr_err("acpuclock: Unable to increase ACPU VDD.\n"); */
+				pr_err("acpuclock: Unable to increase ACPU VDD from %d to %d setting rate to %d.\n", cur->vdd, next->vdd, (int) rate);
 				mutex_unlock(&drv_state.lock);
 				return -EINVAL;
 			}
@@ -354,7 +360,9 @@ int acpuclk_set_rate(unsigned long rate,
 		/* Drop VDD level if we can. */
 		if (next->vdd < cur->vdd) {
 			if (acpuclk_set_vdd_level(next->vdd))
-				pr_err("acpuclock: Unable to drop ACPU VDD.\n");
+			/*	pr_err("acpuclock: Unable to drop ACPU VDD.\n");*/
+				pr_err("acpuclock: Unable to drop ACPU VDD from %d to %d setting rate to %d.\n", cur->vdd, next->vdd, (int) rate);
+                               
 		}
 		mutex_unlock(&drv_state.lock);
 	}
@@ -410,15 +418,15 @@ static void __init acpuclk_init(void)
 		BUG();
 	}
 
-	/* Move to 768MHz for boot, which is a safe frequency
+	/* Move to 883MHz for boot, which is a safe frequency
 	 * for all versions of Scorpion at the moment.
 	 */
 	speed = acpu_freq_tbl;
 	for (;;) {
-		if (speed->acpu_khz == 768000)
+		if (speed->acpu_khz == 883200)
 			break;
 		if (speed->acpu_khz == 0) {
-			pr_err("acpuclk_init: cannot find 768MHz\n");
+			pr_err("acpuclk_init: cannot find 883MHz\n");
 			BUG();
 		}
 		speed++;
diff -rupN linux-2.6.34-rc3-n1/arch/arm/mach-msm/board-mahimahi-audio.c linux-2.6.34-rc3-n1-ocuv/arch/arm/mach-msm/board-mahimahi-audio.c
--- linux-2.6.34-rc3-n1/arch/arm/mach-msm/board-mahimahi-audio.c	2010-04-25 00:03:39.589716909 +0930
+++ linux-2.6.34-rc3-n1-ocuv/arch/arm/mach-msm/board-mahimahi-audio.c	2010-04-25 06:04:56.000000000 +0930
@@ -34,30 +34,30 @@ static struct mutex mic_lock;
 static struct mutex bt_sco_lock;
 
 static struct q6_hw_info q6_audio_hw[Q6_HW_COUNT] = {
-	[Q6_HW_HANDSET] = {
-		.min_gain = -2000,
-		.max_gain = 0,
-	},
-	[Q6_HW_HEADSET] = {
-		.min_gain = -2000,
-		.max_gain = 0,
-	},
-	[Q6_HW_SPEAKER] = {
-		.min_gain = -1500,
-		.max_gain = 0,
-	},
-	[Q6_HW_TTY] = {
-		.min_gain = -2000,
-		.max_gain = 0,
-	},
-	[Q6_HW_BT_SCO] = {
-		.min_gain = -2000,
-		.max_gain = 0,
-	},
-	[Q6_HW_BT_A2DP] = {
-		.min_gain = -2000,
-		.max_gain = 0,
-	},
+        [Q6_HW_HANDSET] = {
+                .min_gain = -1500,
+                .max_gain = 1100,
+        },
+        [Q6_HW_HEADSET] = {
+                .min_gain = -1500,
+                .max_gain = 1100,
+        },
+        [Q6_HW_SPEAKER] = {
+                .min_gain = -1500,
+                .max_gain = 1100,
+        },
+        [Q6_HW_TTY] = {
+                .min_gain = -1500,
+                .max_gain = 1100,
+        },
+        [Q6_HW_BT_SCO] = {
+                .min_gain = -1500,
+                .max_gain = 1100,
+        },
+        [Q6_HW_BT_A2DP] = {
+                .min_gain = -1500,
+                .max_gain = 1100,
+        },
 };
 
 void mahimahi_headset_enable(int en)
diff -rupN linux-2.6.34-rc3-n1/arch/arm/mach-msm/board-mahimahi.c linux-2.6.34-rc3-n1-ocuv/arch/arm/mach-msm/board-mahimahi.c
--- linux-2.6.34-rc3-n1/arch/arm/mach-msm/board-mahimahi.c	2010-04-25 00:03:39.589716909 +0930
+++ linux-2.6.34-rc3-n1-ocuv/arch/arm/mach-msm/board-mahimahi.c	2010-04-25 06:04:56.000000000 +0930
@@ -311,13 +311,13 @@ static struct android_pmem_platform_data
 	.cached		= 1,
 };
 
-static struct android_pmem_platform_data android_pmem_camera_pdata = {
+/*static struct android_pmem_platform_data android_pmem_camera_pdata = {
 	.name		= "pmem_camera",
 	.start		= MSM_PMEM_CAMERA_BASE,
 	.size		= MSM_PMEM_CAMERA_SIZE,
 	.no_allocator	= 1,
 	.cached		= 1,
-};
+};*/
 
 static struct platform_device android_pmem_mdp_device = {
 	.name		= "android_pmem",
@@ -335,13 +335,13 @@ static struct platform_device android_pm
 	},
 };
 
-static struct platform_device android_pmem_camera_device = {
+/*static struct platform_device android_pmem_camera_device = {
 	.name		= "android_pmem",
 	.id		= 2,
 	.dev		= {
 		.platform_data = &android_pmem_camera_pdata,
 	},
-};
+};*/
 
 static struct resource ram_console_resources[] = {
 	{
@@ -437,8 +437,8 @@ static struct regulator_init_data tps650
 	{
 		.constraints = {
 			.name = "dcdc1", /* VREG_MSMC2_1V29 */
-			.min_uV = 1000000,
-			.max_uV = 1300000,
+			.min_uV = 950000,
+			.max_uV = 1275000,
 			.valid_ops_mask = REGULATOR_CHANGE_VOLTAGE,
 		},
 		.consumer_supplies = tps65023_dcdc1_supplies,
@@ -838,7 +838,7 @@ static struct platform_device *devices[]
 	&android_usb_device,
 	&android_pmem_mdp_device,
 	&android_pmem_adsp_device,
-	&android_pmem_camera_device,
+	/*&android_pmem_camera_device,*/
 	&msm_kgsl_device,
 	&msm_device_i2c,
 	&capella_cm3602,
diff -rupN linux-2.6.34-rc3-n1/arch/arm/mach-msm/board-mahimahi.h linux-2.6.34-rc3-n1-ocuv/arch/arm/mach-msm/board-mahimahi.h
--- linux-2.6.34-rc3-n1/arch/arm/mach-msm/board-mahimahi.h	2010-04-25 00:03:39.597720073 +0930
+++ linux-2.6.34-rc3-n1-ocuv/arch/arm/mach-msm/board-mahimahi.h	2010-04-25 06:04:56.000000000 +0930
@@ -42,11 +42,12 @@
 #define MSM_PMEM_ADSP_BASE	0x32000000
 #define MSM_PMEM_ADSP_SIZE	0x02900000
 
-#define MSM_PMEM_CAMERA_BASE	0x34900000
-#define MSM_PMEM_CAMERA_SIZE	0x00800000
+//Extends highmem by 8MB
+/*#define MSM_PMEM_CAMERA_BASE	0x34900000
+#define MSM_PMEM_CAMERA_SIZE	0x00800000*/
 
-#define MSM_HIGHMEM_BASE	0x35100000
-#define MSM_HIGHMEM_SIZE	0x0AF00000
+#define MSM_HIGHMEM_BASE	0x34900000
+#define MSM_HIGHMEM_SIZE	0x0B700000
 
 #define MAHIMAHI_GPIO_PS_HOLD		25
